src/main/java/com/opentable/kafka/builders/KafkaConsumerBuilder.java

```diff
@@ -113,12 +139,12 @@ public KafkaConsumerBuilder(Map<String, Object> prop, EnvironmentProvider enviro
      * @param valDeSer value deserializer
      * @return this
      */
-    public KafkaConsumerBuilder<K, V> withDeserializers(Class<? extends Deserializer<K>> keyDeSer, Class<? extends Deserializer<V>> valDeSer) {
+    public <K2, V2> KafkaConsumerBuilder<K2, V2> withDeserializers(Class<? extends Deserializer<K2>> keyDeSer, Class<? extends Deserializer<V2>> valDeSer) {
         this.keyDe = keyDeSer;
         this.valueDe = valDeSer;
         this.keyDeserializerInstance = null;
         this.valueDeserializerInstance = null;
-        return this;
+        return new KafkaConsumerBuilder<>(this);
```

Comment:
Nope, hate this.

Reply:
I've removed cloning. If this was a concern. ;(